### PR TITLE
CON-146: Test and document not autoenabling readers

### PR DESCRIPTION
**Pull request by alexcamposruiz**
Also changed how errors in entity creation are reported: they are an rti.Error now, and they include the C error messages

### CODE REVIEW

File: test/python/test_rticonnextdds_input.py [hunk 1]
```diff
@@ -60,3 +61,24 @@ def test_reader_native_call(self, rtiInputFixture):
...
+  def test_no_autoenable(self):
+    with open_test_connector("MyParticipantLibrary::TestNoAutoenableSubscriber") as connector:
+      output = connector.get_output("TestPublisher::TestWriter")
+      connector.wait(1000) # TODO
+
+      # The input is not enabled yet because the subscriber sets
+      # autoenable_created_entities to false
+      output.instance['x'] = 1
+      output.write()
+
```
Comment (samuelraeburn) [inline test/python/test_rticonnextdds_input.py]: You could call connector.wait() and check that a timeout error is raised

File: docs/connector.rst [hunk 2]
```diff
@@ -73,6 +73,16 @@ Once you have created a ``Connector`` instance, :meth:`Connector.get_output()`
 returns the :class:`Output` that allows writing data, and :meth:`Connector.get_input()`
 returns the :class:`Input` that allows reading data.
 
+.. note::
+
+  If the *domain_participant* you load contains both *data_writers* (Outputs) and
+  *data_readers* (Inputs) for the same topics, when you write data, the Inputs
```
Comment (samuelraeburn) [inline docs/connector.rst]: for the same topics -> which match (are associated with the same topic and have compatible QoS)

File: docs/connector.rst [hunk 3]
```diff
@@ -73,6 +73,16 @@ Once you have created a ``Connector`` instance, :meth:`Connector.get_output()`
...
+  will receive the data even before you call :meth:`Connector.get_input()`.
+  To avoid that, you can configure the *subscriber* that contains the
+  *data_reader* with ``<subscriber_qos>/<entity_factory>/<autoenable_created_entities>``
+  set to ``false``. In that case, the Inputs will only receive data after you
+  call :meth:`Connector.get_input()`.
+
```
Comment (samuelraeburn) [inline docs/connector.rst]: Do you think we should mention here the bigger issue of the un-wanted data_reader's queue filling up and blocking the writer in reliable communication?

File: test/python/test_rticonnextdds_input.py (hunk 1 above)
Comment (alexcamposruiz) [reply test/python/test_rticonnextdds_input.py]: Checking that only the second sample is received is enough, and doesn't add an additional sleep time.

File: docs/connector.rst (hunk 3 above)
Comment (alexcamposruiz) [reply docs/connector.rst]: I didn't want to get into everything that can go wrong. That's the most obvious side effect, but there could be more. I think I'll leave it to the reader to figure it out.

Comment (alexcamposruiz) [reply docs/connector.rst]: The human reading the documentation, that is.

**Pull request merged**